server/src/test/java/org/elasticsearch/action/support/IndicesOptionsTests.java

```diff
@@ -59,14 +60,17 @@ public void testSerialization() throws Exception {
             StreamInput streamInput = output.bytes().streamInput();
             IndicesOptions indicesOptions2 = IndicesOptions.readIndicesOptions(streamInput);
 
-            assertThat(indicesOptions2.ignoreUnavailable(), equalTo(indicesOptions.ignoreUnavailable()));
-            assertThat(indicesOptions2.allowNoIndices(), equalTo(indicesOptions.allowNoIndices()));
-            assertThat(indicesOptions2.expandWildcardsOpen(), equalTo(indicesOptions.expandWildcardsOpen()));
-            assertThat(indicesOptions2.expandWildcardsClosed(), equalTo(indicesOptions.expandWildcardsClosed()));
-            assertThat(indicesOptions2.expandWildcardsHidden(), equalTo(indicesOptions.expandWildcardsHidden()));
+            MatcherAssert.assertThat(indicesOptions2.ignoreUnavailable(), equalTo(indicesOptions.ignoreUnavailable()));
```

Comment:
Is there a reason you do this instead of `assertEquals(..., ...)`? `ESTestCase` extends `org.junit.Assert`, which in turn implements `assertEquals`. That seems like a cleaner approach to me but maybe I'm missing something.

Reply:
It was marked as deprecated, this is why I did that:

> Deprecated use org.hamcrest.MatcherAssert.assertThat()

Let me confirm that my setup is correct, maybe something is wrongly linked?

Reply:
Yeah for me it also says that the original way is deprecated. My suggestion was that we do `assertEquals(indicesOptions2.ignoreUnavailable(), indicesOptions.ignoreUnavailable())`. That way, we don't have to import `org.hamcrest.MatcherAssert` (and it's a tiny bit shorter as we don't have to wrap the second argument with `equalTo(...)`). I would assume both approaches work the same, but I don't know that for sure.

Reply:
I will convert this to draft, and look into this a bit more.